osf/management/commands/sync_doi_metadata.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from osf.models import Identifier
+
+from framework.celery_tasks import app
+from framework.celery_tasks.handlers import enqueue_task
+
+logger = logging.getLogger(__name__)
+
+
+@app.task(name='osf.management.commands.sync_doi_metadata', max_retries=5, default_retry_delay=60)
+def sync_identifier_doi(identifier):
+    identifier.referent.request_identifier_update('doi')
+    identifier.save()
+    logger.info(f' doi update for {identifier.value} complete')
+
+
+def sync_doi_metadata(modified_date, batch_size=100, dry_run=True):
+    identifiers = Identifier.objects.filter(
+        category='doi',
+        deleted__isnull=True,
+        modified__lte=modified_date,
+        object_id__isnull=False,
+    )[:batch_size]
+    logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
+                f'{identifiers.count()} identifiers to mint')
+
+    for identifier in identifiers:
+        if not dry_run:
```

Comment:
Let's make sure that the referent isn't deleted or withdrawn (or spam? or should deleted catch that?) before we spin up the task.

Reply:
The client sends a withdraw request when it has a deleted/private node, I was hoping that would update some of those old EZID DOIs, and potentially other's that might have randomly not been contacted to network issues.

Reply:
Yep. `not identifier.referent.is_retracted` should probably also be checked below, too.